### PR TITLE
Merge Array test

### PR DESCRIPTION
**Pull request by ZiheZhou1101**

### CODE REVIEW

File: reactExercise/solutions/zihe/ex10/src/App.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { useState } from "react"
+
+export const mergeArray=(arrayA,arrayB)=>{
+  const A=[...arrayA,...arrayB]
+  if(A.length===0) return []
+  let mergeAB=[]
+  mergeAB.push(A[0])
+  for(let i=1;i<A.length;i++) {
```
Comment (sccdmaomao) [inline reactExercise/solutions/zihe/ex10/src/App.js:8]: 这里为啥要放第一个element 然后再 从index 1开始loop

Comment (sccdmaomao) [reply reactExercise/solutions/zihe/ex10/src/App.js:8]: 我猜你是觉得第一个不可能是重复 所以直接放进去.  这个逻辑应该不需要才对

File: reactExercise/solutions/zihe/ex10/src/App.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+         mergeAB.splice(j,0,A[i])
+          break
+        }
+     }
+   }
+   return mergeAB
+}
+
+function App() {
+  const arrayA=[1,5,8,3,1,5,2]
```
Comment (sccdmaomao) [inline reactExercise/solutions/zihe/ex10/src/App.js:25]: 这俩是测试用的数据吧  可以删掉

File: reactExercise/solutions/zihe/ex10/src/App.test.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import { mergeArray } from "./App";
+import App from "./App";
+import '@testing-library/jest-dom'
+import { fireEvent,render,screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+describe('MergeArray',()=>{
```
Comment (sccdmaomao) [inline reactExercise/solutions/zihe/ex10/src/App.test.js:7]: 👍

File: reactExercise/solutions/zihe/ex10/src/App.test.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    test('common case',()=>{
```
Comment (sccdmaomao) [inline reactExercise/solutions/zihe/ex10/src/App.test.js:8]: 哈哈哈 我写的common case, base case, edge case是来做demo的
自己写code的话 可以尽量描述测试的是什么   比如说  
- empty array merge
- all duplicate merge

**Review by sccdmaomao — APPROVED** [review]

**Comment by sccdmaomao** [comment]
你的mergeArray虽然写出来了 但是我觉得不够clean 和efficient
可能还要多练习练习 algorithm 和data structure

File: reactExercise/solutions/zihe/ex10/src/App.js (hunk 1 above)
Comment (ZiheZhou1101) [reply reactExercise/solutions/zihe/ex10/src/App.js:8]: if(A[i]>mergeAB[mergeAB.length-1]) {
      mergeAB.push(A[i])
      continue
    }为了让mergeAB.length-1是个非负数
